### DISCUSSION TITLE
What is a good template engine to be used with HTMX?

### DISCUSSION THREAD

**Discussion by sqllyw** [bigskysoftware/htmx#2377] [Q&A]
hi,
now I'm using handlebar, for every page I have one main template file with several small template files that contains fragments, any template that can combine the main page with fragments? thanks

**Comment by CaliViking** [comment]
@sqllyw - I use Jinja2 - it is great. It supports both inheritance (through blocks) and composition (through macros). It has great logic to make the rendering conditional and repetitive.
What is your backend? You need to choose one that is supported.
